Backend/dev/mock-servers/services/google.py

```
@@ -13,9 +13,31 @@
+    return os.environ.get("OSRM_HOST", "http://localhost:5001")
+
+
+def _osrm_table(coords):
+    # coords = list of "lat,lon" strings
+    # OSRM expects lon,lat order
+    coord_str = ";".join(
+        f"{lon},{lat}" for lat, lon in (c.split(",") for c in coords)
+    )
+    url = f"{_osrm_base_url()}/table/v1/driving/{coord_str}?annotations=duration,distance"
+    try:
+        with urllib.request.urlopen(url, timeout=2) as resp:
```

Ruff (0.15.20)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)